app/use_cases/process_enrolments.py

```diff
@@ -0,0 +1,39 @@
+from pydantic import BaseModel
+
+from app.repositories.enrolment_repo import EnrolmentRepo
+from app.repositories.s3_enrolment_repo import state_change_valid
+from app.requests.enrolment_requests import ProcessEnrolmentRequest
+from app.responses import ResponseFailure, ResponseSuccess, SuccessType
+
+
+class ProcessEnrolments(BaseModel):
+    enrolment_repo: EnrolmentRepo
+
+    class Config:
+        # Pydantic will complain if something (course_repo) is defined
+        # as having a non-BaseModel type (e.g. an ABC). Setting this ensures
+        # that it will just check that the value isinstance of this class.
+        arbitrary_types_allowed = True
+
+    def execute(self, process_enrolment_request: ProcessEnrolmentRequest):
+        try:
+            new_state = process_enrolment_request.new_state
+            enrolment_id = process_enrolment_request.enrolment_request_id
+            enrolment = self.enrolment_repo.get_enrolment_by_id(enrolment_id)
+            if not enrolment:
```

Comment:
should we not implement these checks in s3_enrolment_repo.py. just thinking

Reply:
Okay, I'll add enrolment_auth_exists() method, to S3 Enrolment Repo and it will return True/False